src/main/java/rotateit/service/EventPublisher.java

```diff
@@ -0,0 +1,21 @@
+package rotateit.service;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.event.ApplicationEventMulticaster;
+import org.springframework.stereotype.Component;
+import rotateit.domain.Event;
+
+@Component
```

Comment:
you already defined it as a bean in `ApplicationContext` no need for `@Component`, we are not using component scanning